akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -184,13 +184,14 @@ object Source {
    * Streams the elements of the given future source once it successfully completes.
    * If the future fails the stream is failed.
    */
-  def fromFutureSource[T, M](future: Future[Graph[SourceShape[T], M]]): javadsl.Source[T, Future[M]] = new Source(scaladsl.Source.fromFutureSource(future))
+  def fromFutureSource[T, M](future: Future[_ <: Graph[SourceShape[T], M]]): javadsl.Source[T, Future[M]] = new Source(scaladsl.Source.fromFutureSource(future))
```

Comment:
Future is variant, but this is the Java API, it needs to be variant at the call site.